Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppProtectionPolicyAndroid/MSFT_IntuneAppProtectionPolicyAndroid.psm1

```diff
@@ -1125,52 +1241,120 @@ function set-AppsHash
     return $AppsHash
 }
 
+Function set-ManagedBrowserValues
+{
+    param (
+        [string]$ManagedBrowser,
+        [switch]$ManagedBrowserToOpenLinksRequired,
+        [string]$CustomBrowserDisplayName,
+        [string]$CustomBrowserPackageId
+    )
+
+   # via the gui there are only 3 possible configs:
+   # edge - edge, true, empty id strings
+   # any app - not configured, false, empty strings
+   # unmanaged browser not configured, true, strings must not be empty
+
+   write-host 'Setting Managed Browser Properties'
+
+   if (!$ManagedBrowserToOpenLinksRequired)
+   {
+        write-host 'Setting Managed Browser to Any App'
+        $ManagedBrowser = 'notConfigured'
+        $ManagedBrowserToOpenLinksRequired = $false
+        $CustomBrowserDisplayName = ''
+        $CustomBrowserPackageId = ''
+
+   }
+   else
+   {
+        if(($CustomBrowserDisplayName -ne '') -and ($CustomBrowserPackageId -ne ''))
+        {
+            write-host 'Setting Managed Browser to Custom Browser'
+            $ManagedBrowser = 'notConfigured'
+            $ManagedBrowserToOpenLinksRequired = $true
+            $CustomBrowserDisplayName = $CustomBrowserDisplayName
+            $CustomBrowserPackageId = $CustomBrowserPackageId
+        }
+        else
+        {
+            write-host 'Setting Managed Browser to Microsoft Edge'
+            $ManagedBrowser = 'microsoftEdge'
+            $ManagedBrowserToOpenLinksRequired = $true
+            $CustomBrowserDisplayName = ''
+            $CustomBrowserPackageId = ''
+        }
+
+   }
+
+
+    $ManagedBrowserHash = @{
+        'ManagedBrowser' = $ManagedBrowser;
+        'ManagedBrowserToOpenLinksRequired' = $ManagedBrowserToOpenLinksRequired;
+        'CustomBrowserDisplayName' = $CustomBrowserDisplayName;
+        'CustomBrowserPackageId' = $CustomBrowserPackageId
+    }
+
+    return $ManagedBrowserHash
+
```

Comment:
Remove a few empty lines to keep it clean. Picky, I know ;)

Reply:
@NikCharlebois No worries, I've made those amendments now - let me know if there's anything else needed.  thanks